instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java

```diff
@@ -0,0 +1,283 @@
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12]"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12].Tasklet"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12]"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12].Tasklet"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(3))));
+  }
+
+  @Test
+  void shouldTraceJobWithDecision() {
+    runner.runJob("decisionJob");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob decisionJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepStart")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepStart.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepLeft")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepLeft.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(3))));
+  }
+
+  @Test
+  void shouldTracePartitionedJob() {
+    runner.runJob("partitionedJob");
+
+    testing.waitAndAssertTraces(
+        trace -> {
+          int index = 2;
```

Comment:
this is only referenced twice, is there a reason we need to declare it as a var?